src/compiler/checker.ts

```diff
@@ -31443,7 +31734,7 @@ namespace ts {
                     const type = checkNewTargetMetaProperty(node);
                     return isErrorType(type) ? errorType : createNewTargetExpressionType(type);
                 default:
-                    Debug.assertNever(node.keywordToken);
+                    throw new Error("never");

```

Comment:
Likewise, `Debug.assertNever` is ansuring `node.keywordToken` is exhaustively handled - it shouldn't be removed.